src/com/company/gamelogger/LogConfFactory.java

```diff
@@ -0,0 +1,184 @@
+package com.company.gamelogger;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.ConsoleAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationFactory;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.builder.api.*;
+import org.apache.logging.log4j.core.config.builder.impl.BuiltConfiguration;
+import org.apache.logging.log4j.core.config.Order;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+
+import java.net.URI;
+
+import static com.company.gamelogger.LogOptions.*;
+
+@Plugin(name = "LogConfFactory", category = ConfigurationFactory.CATEGORY)
+@Order(50)
+public class LogConfFactory extends ConfigurationFactory {
+    private static Configuration createConfiguration(final String name, ConfigurationBuilder<BuiltConfiguration> builder) {
+        builder.setConfigurationName(name);
+        builder.setStatusLevel(LEVEL);
+
+        String logFilePath = FILE_PATH + "/" + FILE_NAME;
+        boolean debugging = LEVEL.equals(Level.DEBUG) || LEVEL.equals(Level.ALL);
```

Comment:
This is more correct:
boolean debugging = LEVEL.getlevel() <= Level.DEBUG;

Reply:
See new commit!